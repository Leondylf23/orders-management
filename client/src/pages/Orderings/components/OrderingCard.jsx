import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import classes from '../style.module.scss';

const OrderingCard = ({ data, onClickDetail, isBusiness }) => {
  const [status, setStatus] = useState(null);

  useEffect(() => {
    if (data?.status) {
      switch (data?.status) {
        case 'WAITING':
          setStatus({
            text: <FormattedMessage id="status_waiting" />,
            color: 'yellow',
          });
          break;
        case 'FAILED':
          setStatus({
            text: <FormattedMessage id="status_failed" />,
            color: 'red',
          });
          break;
        case 'SUCCESS':
          setStatus({
            text: <FormattedMessage id="status_success" />,
            color: 'green',
          });
          break;
      }
    }
  }, [data]);

  return (
    <div className={classes.orderingCard} onClick={() => onClickDetail(data?.id)}>
      <img className={classes.cardImage} src={data?.imageUrl} alt="Img Failed" />
      <div className={classes.cardContent}>
        <h4 className={classes.orderCode}>{data?.transactionCode}</h4>
        <h4 className={classes.title}>{data?.title}</h4>
        {isBusiness && <p className={classes.customerName}>{data?.customer}</p>}
        <div className={classes.statusContainer} data-item={status?.color}>
          <h5 className="">{status?.text}</h5>
        </div>
      </div>
    </div>
  );
};

OrderingCard.propTypes = {
  data: PropTypes.object.isRequired,
  onClickDetail: PropTypes.func.isRequired,
  isBusiness: PropTypes.bool,
};

export default OrderingCard;
