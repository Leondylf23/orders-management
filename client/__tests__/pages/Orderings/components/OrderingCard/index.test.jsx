import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
// import RouterDom from 'react-router-dom';

import OrderingCard from '@pages/Orderings/components/OrderingCard';
import store from '@store';
import Language from '@containers/Language';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

jest.mock('reselect', () => ({
  ...jest.requireActual('reselect'),
}));

const ParentComponent = (children) => (
  <Provider store={store}>
    <Language>{children}</Language>
  </Provider>
);

// const navigate = jest.fn().mockName('navigate');

describe('Ordering Card', () => {
  // beforeEach(() => {
  //     jest.spyOn(RouterDom, 'useNavigate').mockImplementation(() => navigate);
  // });

  test('Rendered', () => {
    const { getByTestId } = render(ParentComponent(<OrderingCard />));
    const orderingCard = getByTestId('ordering-card-page');
    expect(orderingCard).toBeInTheDocument();
  });

  test('Rendered with correct value if role is customer', () => {
    const testData = {
      imageUrl: 'url://test-image/cust',
      transactionCode: 'trx/data-test/00-00-00/cust',
      title: 'test data customer',
    };

    const { getByTestId } = render(ParentComponent(<OrderingCard data={testData} />));

    const imageElement = getByTestId('ordering-card-page-img');
    expect(imageElement.src).toBe(testData?.imageUrl);

    const transactionCode = getByTestId('ordering-card-page-transaction-code');
    expect(transactionCode.textContent).toBe(testData?.transactionCode);

    const title = getByTestId('ordering-card-page-title');
    expect(title.textContent).toBe(testData?.title);
  });

  test('Rendered with correct value if role is business', () => {
    const testData = {
      imageUrl: 'url://test-image/business',
      transactionCode: 'trx/data-test/00-00-00/business',
      title: 'test data business',
      customer: 'customer business test',
    };

    const { getByTestId } = render(ParentComponent(<OrderingCard data={testData} isBusiness />));

    const imageElement = getByTestId('ordering-card-page-img');
    expect(imageElement.src).toBe(testData?.imageUrl);

    const transactionCode = getByTestId('ordering-card-page-transaction-code');
    expect(transactionCode.textContent).toBe(testData?.transactionCode);

    const title = getByTestId('ordering-card-page-title');
    expect(title.textContent).toBe(testData?.title);

    const customer = getByTestId('ordering-card-page-customer');
    expect(customer.textContent).toBe(testData?.customer);
  });

  test('Should match with snapshot', () => {
    const orderingCard = render(ParentComponent(<OrderingCard />));
    expect(orderingCard).toMatchSnapshot();
  });
});
