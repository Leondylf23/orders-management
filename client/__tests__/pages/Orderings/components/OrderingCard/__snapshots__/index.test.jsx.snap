// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ordering Card Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="orderingCard"
        data-testid="ordering-card-page"
      >
        <img
          alt="Img Failed"
          class="cardImage"
          data-testid="ordering-card-page-img"
        />
        <div
          class="cardContent"
        >
          <h4
            class="orderCode"
            data-testid="ordering-card-page-transaction-code"
          />
          <h4
            class="title"
            data-testid="ordering-card-page-title"
          />
          <div
            class="footer"
          >
            <div
              class="statusContainer"
            >
              <h5
                class="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="orderingCard"
      data-testid="ordering-card-page"
    >
      <img
        alt="Img Failed"
        class="cardImage"
        data-testid="ordering-card-page-img"
      />
      <div
        class="cardContent"
      >
        <h4
          class="orderCode"
          data-testid="ordering-card-page-transaction-code"
        />
        <h4
          class="title"
          data-testid="ordering-card-page-title"
        />
        <div
          class="footer"
        >
          <div
            class="statusContainer"
          >
            <h5
              class="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
